src/Schema/Comparator.php

```diff
@@ -57,7 +57,7 @@ public function __call(string $method, array $args): SchemaDiff
             throw new BadMethodCallException(sprintf('Unknown method "%s"', $method));
         }
 
-        return $this->doCompareSchemas(...$args);
+        return (new self())->doCompareSchemas(...$args);
```

Comment:
This practically means "disable all platform-aware schema diffing", which is potentially not the correct root cause, but at least brings us back to the `3.3.0` behavior, for now.

Reply:
In practice, it seems like nothing in platform-aware diffing is affecting anything in the test suite (all green, even if I effectively killed the feature): kill it before it lays eggs, and becomes more complex than needed?

Reply:
Potentially better solution: keep the test from this patch, and add the commits from #5224 on top of it.